client/src/components/application/NonlinearStepper/index.js

```diff
@@ -162,17 +160,19 @@ export default function HorizontalNonLinearStepper({ userID, UserData }) {
           <> </>
         )}
         <ThemeProvider theme={Theme}>
-          <Stepper nonLinear activeStep={activeStep}>
-            {steps.map((label, index) => (
-              <Step key={label}>
-                <StepButton onClick={handleStep(index)}>{label}</StepButton>
-              </Step>
-            ))}
-          </Stepper>
           <div>
+            <Stepper nonLinear activeStep={activeStep}>
+              {steps.map((label, index) => (
+                <Step key={label}>
+                  <StepButton onClick={handleStep(index)}>{label}</StepButton>
```

Comment:
I think your onClick handler won't quite work right, because what is happening is that you are calling `handleStep(index)` and passing its return value (which is probably undefined) as the `onClick` prop instead of actually passing in a function to `onClick`.  Instead, you might need to rewrite it like this:
`onClick={() => handleStep(index)}`